components/playlist/playlist_download_request_manager.cc

```diff
@@ -161,6 +161,10 @@ void PlaylistDownloadRequestManager::GetMedia(content::WebContents* contents) {
 void PlaylistDownloadRequestManager::OnGetMedia(
     base::WeakPtr<content::WebContents> contents,
     base::Value value) {
+  base::ScopedClosureRunner scoped_closure_runner(
+      base::BindOnce(&PlaylistDownloadRequestManager::FetchPendingRequest,
+                     weak_factory_.GetWeakPtr()));
```

Comment:
how about splitting this into two functions and call `FetchPendingRequest` regardless of the internal stuff?
`ScopedClosureRunner` looks pretty vague. having something like this seems cleaner:
```
void PlaylistDownloadRequestManager::OnGetMedia(
    base::WeakPtr<content::WebContents> contents,
    base::Value value) {
  Send/Process/Handle/???FoundMedia(contents, std::move(value));
  FetchPendingRequest();
}
```

Reply:
Yeah, that'd be more like a "SOLID" function.